From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BillyGalbreath <blake.galbreath@gmail.com>
Date: Thu, 15 Dec 2022 02:03:38 -0600
Subject: [PATCH] Fix MC-258430 Camel LastPoseTime

This isn't a proper fix, but just a temporary workaround by
resetting the LastPoseTime when a player mounts a camel.

diff --git a/src/main/java/net/minecraft/world/entity/animal/camel/Camel.java b/src/main/java/net/minecraft/world/entity/animal/camel/Camel.java
index f9d77ad2493b9ba67e8f45ce248d29a8e9ab7b80..1074d6bda611feb60230168774cc3f8b278c8a38 100644
--- a/src/main/java/net/minecraft/world/entity/animal/camel/Camel.java
+++ b/src/main/java/net/minecraft/world/entity/animal/camel/Camel.java
@@ -85,6 +85,12 @@ public class Camel extends AbstractHorse implements PlayerRideableJumping, Rider
     public int getPurpurBreedTime() {
         return this.level.purpurConfig.camelBreedingTicks;
     }
+
+    @Override
+    protected void doPlayerRide(Player player) {
+        resetLastPoseChangeTick(this.level.getGameTime());
+        super.doPlayerRide(player);
+    }
     // Purpur end
 
     @Override
